baseband/mark4/base.py

```diff
@@ -374,7 +374,7 @@ def read(self, count=None, fill_value=0., out=None):
             frame_nr, sample_offset = divmod(self.offset,
                                              self.samples_per_frame)
             if frame_nr != self._frame_nr:
-                self._read_frame()
+                self._read_frame(fill_value)
 
             data = self._frame.data
```

Comment:
Actually, I was wrong below, since here it just gets `_frame.data` again instead of using `_frame_data`. Since neither payload nor frame cache the decoded data, I think we should use `_frame_data` here.

Reply:
Removing that prevents decoding the first frame by the stream reader.  We could add `self._frame_data = self._frame.data` in `__init__`, but I think that keeping `data = self._frame.data` in `read` while removing it in `_read_frame` is a more elegant (if less efficient) solution and makes the code more symmetric with VDIF.  See further comments below.